content/docs/main/resiliency/retry/per-try-timeout.md

```diff
@@ -0,0 +1,288 @@
+---
+title: Per-try timeout
+weight: 20
+description:
+---
+
+Set separate timeouts for retries. 
+
+## About per-try timeouts
+
+The per-retry timeout allows you to set a timeout for retried requests. If the timeout expires, Envoy cancels the retry attempt and immediately retries on another upstream host. 
+
+By default, Envoy has a default overall request timeout of 15 seconds. A request timeout represents the time Envoy waits for the entire request to complete, including retries. Without a per-try timeout, retries might take longer than the overall request timeout, and therefore might not be executed as the request times out before the retry attempts can be performed. You can configure a larger [request timeout]({{< link-hextra path="/resiliency/timeouts/request/" >}}) to account for this case. However, you can also define timeouts for each retry so that you can protect against slow retry attempts from consuming the entire request timeout.
+
+<!--
+When per-try timeouts are enabled, Envoy returns the `x-envoy-upstream-rq-per-try-timeout-ms` header in the response.  -->
+Note that if you configured a global timeout, the per-try timeout must be less than the global timeout.
+
+Per-try timeouts can be configured on an HTTPRoute directly. To enable per-try timeouts on a Gateway listener level, use a {{< reuse "docs/snippets/trafficpolicy.md" >}} instead. 
+
+{{< callout type="warning" >}} 
+{{< reuse "docs/versions/warn-experimental.md" >}}
+{{< /callout >}}
+
+{{< callout >}}
+{{< reuse "docs/snippets/proxy-kgateway.md" >}}
```

Comment:
Can we add this to the agentgateway limitations here? https://kgateway.dev/docs/main/agentgateway/about/#more-considerations